### PR TITLE
Clarified usage of asset keys

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):** #198

**Proposed Changes:**

1. Removed asset types (replaced by role).
2. Explained how to use the asset keys.

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: item-spec/item-spec.md [hunk 1]
```diff
@@ -183,14 +187,6 @@ Like the Link `rel` field, the `role` field can be given any value, however here
 | overview  | An asset that represents a possibly larger view than the thumbnail of the Item , for example, a true color composite of multi-band data. |
 | metadata  | A metadata sidecar file describing the data in this item, for example the Landsat-8 MTL file. |
 
-#### Asset types
-
-The following types are common for assets and are used as the key for the Asset Object:
-
-| Type      | Description |
-| --------- | ----------- |
-| thumbnail | STRONGLY RECOMMENDED. A downsampled image of the core asset, for direct display online in a web page or interactive search application. Even assets that are less easily translated in to a visual image should provide some visual representation, that users can browse. For example a SAR asset can render an elevation mask or hillshade for display. If at all possible it should be included for a better user experience in searching data. |
```
Comment (cholmes) [inline item-spec/item-spec.md]: I'm cool with moving this out, but I think we need to add that it's 'strongly recommended' in the 'roles' section. After that's done I'm up to approve this.

Comment (m-mohr) [reply item-spec/item-spec.md]: Added it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Removed
 - Removed `version` field in STAC Collections. Use [Version Extension](extensions/version/README.md) instead.
 - Removed `summaries` field from Catalogs. Use Collections instead.
+- Asset Types in Items. Use the asset's `role` instead.
```
Comment (anayeaye) [inline CHANGELOG.md]: It seems like there is a slight chance that there could be confusion about removing Item asset object `type` i.e. media type. Is this more of a 'Fix' to the Item asset object description than a 'Remove'?

Comment (m-mohr) [reply CHANGELOG.md]: I made this clearer now.

File: CHANGELOG.md [hunk 3]
```diff
@@ -37,11 +37,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Removed
 - Removed `version` field in STAC Collections. Use [Version Extension](extensions/version/README.md) instead.
 - Removed `summaries` field from Catalogs. Use Collections instead.
+- Asset Types (pre-defined values for the keys of individual assets, *not* media types) in Items. Use the asset's `role` instead.
```
Comment (anayeaye) [inline CHANGELOG.md]: Thanks for the additional clarification here

**Pull request merged**